core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -180,28 +180,20 @@ public static function indexPage($arrData)
 								->limit(1)
 								->execute($arrSet['checksum'], $arrSet['pid']);
 
-		// Update the URL if the new URL is shorter or the current URL is not canonical
-		if ($objIndex->numRows && $objIndex->url != $arrSet['url'])
+		if ($objIndex->numRows)
 		{
-			if (strpos($arrSet['url'], '?') === false && strpos($objIndex->url, '?') !== false)
+			// The new URL is more canonical (shorter and/or less fragments)
+			if (self::compareUrls($arrSet['url'], $objIndex->url) < 0)
 			{
-				// The new URL is more canonical (no query string)
-				$objDatabase->prepare("DELETE FROM tl_search WHERE id=?")
-							->execute($objIndex->id);
+				self::removeEntry($arrSet['url']);
 
-				$objDatabase->prepare("DELETE FROM tl_search_index WHERE pid=?")
+				$objDatabase->prepare("UPDATE tl_search %s WHERE id=?")
```

Comment:
Why do we need to adjust the whole entry - wouldn't it be enough to alter the url?

Reply:
We could skip `checksum` and `text` I think, but the rest needs to be updated.

Reply:
The `removeEntry()` call would remove a different row, not the one that we want to update.  
I think updating the whole row is less error prone and makes sure that our data is consistent. Therefore I would leave it as is.